shared/lib/confirmation.utils.ts

```diff
@@ -0,0 +1,169 @@
+
+/* eslint-disable jsdoc/require-param, jsdoc/check-param-names */
+
+/** List of signature approval types that support the redesigned confirmation flow */
+const REDESIGN_SIGNATURE_APPROVAL_TYPES = [
+  ApprovalType.EthSignTypedData,
+  ApprovalType.PersonalSign,
+];
+
+/** List of transaction types that support the redesigned confirmation flow for users */
+const REDESIGN_USER_TRANSACTION_TYPES = [
+  TransactionType.contractInteraction,
+  TransactionType.deployContract,
+  TransactionType.tokenMethodApprove,
+  TransactionType.tokenMethodIncreaseAllowance,
+  TransactionType.tokenMethodSetApprovalForAll,
+  TransactionType.tokenMethodTransfer,
+  TransactionType.tokenMethodTransferFrom,
+  TransactionType.tokenMethodSafeTransferFrom,
+  TransactionType.simpleSend,
+];
+
+/** List of transaction types that support the redesigned confirmation flow for developers */
+const REDESIGN_DEV_TRANSACTION_TYPES = [...REDESIGN_USER_TRANSACTION_TYPES];
+
+/**
+ * Determines whether to use the redesigned confirmation flow for a given transaction
+ * based on user settings and developer mode
+ *
+ * @param opts.transactionMetadataType - The type of transaction to check
+ * @param opts.isRedesignedTransactionsUserSettingEnabled - Whether the user has enabled the redesigned flow
+ * @param opts.isRedesignedConfirmationsDeveloperEnabled - Whether developer mode is enabled
+ */
+export function shouldUseRedesignForTransactions({
+  transactionMetadataType,
+  isRedesignedTransactionsUserSettingEnabled,
+  isRedesignedConfirmationsDeveloperEnabled,
+}: {
+  transactionMetadataType?: TransactionType;
+  isRedesignedTransactionsUserSettingEnabled: boolean;
+  isRedesignedConfirmationsDeveloperEnabled: boolean;
+}): boolean {
+  return (
+    shouldUseRedesignForTransactionsUserMode(
+      isRedesignedTransactionsUserSettingEnabled,
+      transactionMetadataType,
+    ) ||
+    shouldUseRedesignForTransactionsDeveloperMode(
+      isRedesignedConfirmationsDeveloperEnabled,
+      transactionMetadataType,
+    )
+  );
+}
+
+/**
+ * Determines whether to use the redesigned confirmation flow for a given signature
+ * based on user settings and developer mode
+ *
+ * @param opts.approvalType - The type of signature approval to check
+ * @param opts.isRedesignedSignaturesUserSettingEnabled - Whether the user has enabled the redesigned flow
+ * @param opts.isRedesignedConfirmationsDeveloperEnabled - Whether developer mode is enabled
+ */
+export function shouldUseRedesignForSignatures({
+  approvalType,
+  isRedesignedSignaturesUserSettingEnabled,
+  isRedesignedConfirmationsDeveloperEnabled,
+}: {
+  approvalType?: ApprovalType;
+  isRedesignedSignaturesUserSettingEnabled: boolean;
+  isRedesignedConfirmationsDeveloperEnabled: boolean;
+}): boolean {
+  const isRedesignedConfirmationsDeveloperSettingEnabled =
+    process.env.ENABLE_CONFIRMATION_REDESIGN === 'true' ||
+    isRedesignedConfirmationsDeveloperEnabled;
```

Comment:
I was confused about how `ENABLE_CONFIRMATION_REDESIGN` was being handled. Thanks for clearing this up and for the helpful tests!